mlflow/tracking/fluent.py

```diff
@@ -686,20 +722,38 @@ def log_metric(key: str, value: float, step: Optional[int] = None) -> None:
                   All backend stores will support values up to length 5000, but some
                   may support larger values.
     :param step: Metric step (int). Defaults to zero if unspecified.
+    :param synchronous: [Experimental] Defaults to True.
+                        Indicates if the metric would be logged in synchronous fashion or not.
+                        When it is True this call would be blocking call and offers immediate
+                        consistency of the metric upon returning.
+                        When this value is set to False, metric would be logged in async
+                        fashion with eventual consistency.
+
+    :return: :py:class:`mlflow.utils.async_logging.run_operations.RunOperations` object.
+             This object, when not None, can be awaited to know if given metric was processed
+             or not.
 
-    .. testcode:: python
         :caption: Example
 
         import mlflow
 
+        # Log a metric
         with mlflow.start_run():
             mlflow.log_metric("mse", 2500.00)
+
+        # Log a metric in async fashion.
+        with mlflow.start_run():
+            mlflow.log_metric("mse", 2500.00, synchronous=False)
     """
     run_id = _get_or_start_run().info.run_id
-    MlflowClient().log_metric(run_id, key, value, get_current_time_millis(), step or 0)
+    return MlflowClient().log_metric(
+        run_id, key, value, get_current_time_millis(), step or 0, synchronous=synchronous
+    )
 
 
-def log_metrics(metrics: Dict[str, float], step: Optional[int] = None) -> None:
+def log_metrics(
+    metrics: Dict[str, float], step: Optional[int] = None, synchronous: Optional[bool] = True
```

Comment:
```suggestion
    metrics: Dict[str, float], step: Optional[int] = None, synchronous: bool = True
```

`Optional` can be removed.